src/app/services/actions/query-action-creators.ts

```diff
@@ -61,42 +99,142 @@ export function runQuery(query: IQuery): Function {
     }
 
     if (response && response.ok) {
-
       status.ok = true;
       status.messageType = MessageBarType.success;
 
       dispatch(setQueryResponseStatus(status));
 
-      return dispatch(queryResponse({
-        body: result,
-        headers: respHeaders
-      }));
-    }
-    else {
-      dispatch(queryResponse({
-        body: result,
-        headers: respHeaders
-      }));
-      return dispatch(setQueryResponseStatus(status));
-    }
+      if (isFileResponse(respHeaders)) {
+        const contentDownloadUrl = await generateResponseDownloadUrl(
+          response,
+          respHeaders
+        );
+        if (contentDownloadUrl) {
+          result = {
+            contentDownloadUrl: contentDownloadUrl,
+          };
+        }
+      }
 
+      return dispatch(
+        queryResponse({
+          body: result,
+          headers: respHeaders,
+        })
+      );
+    } else {
+      if (
+        response.status === 0 &&
+        tokenPresent &&
```

Comment:
**question**: Does this mean this workaround is not available for anonymous users?

Reply:
It's not necessary. There are no redirects happening.